js/datastore.md

```diff
@@ -0,0 +1,566 @@
+				: PostStatus.ACTIVE;
+  }),
+  c => c.rating("gt", 3)
+);
+```
+
+Conditional updates can only be applied to single items and not lists. If you wish to update a list of items you can loop over them and apply conditions one at a time.
+
+## Delete Data
+
+To delete an item simply pass in an instance:
+
+```javascript
+const todelete = await DataStore.query(Post, "123");
+DataStore.delete(todelete);
+```
+
+You can also pass predicate operators to delete multiple items. For example will delete all inactive posts:
+
+```javascript
+await DataStore.delete(Post, c => c.status("eq", PostStatus.INACTIVE));
+```
+
+Additionally you can perform a conditional delete, for instance only delete **if** a post is inactive by passing in an instance of a model:
+
+```javascript
+const todelete = await DataStore.query(Post, "123");
+DataStore.delete(todelete, c => c.status("eq", PostStatus.INACTIVE));
+```
+
+Also to delete all items for a model you can use `Predicates.ALL`:
+
+```javascript
+await DataStore.delete(Post, Predicates.ALL);
+```
+
+## Observe Data
+
+You can subscribe to changes on your Models by using `observe` in the DataStore API. This reacts dynamically to updates of data to the underlying Storage Engine, which could be the result of GraphQL Subscriptions as well as Queries or Mutations that run against the backing AppSync API if you are synchronizing with the cloud.
+
+**Note**: `observe` is async however you should not put `await` in front of it like the other DataStore API methods.
+
+```javascript
+const subscription = DataStore.observe(Post).subscribe(msg => {
+  console.log(msg.model, msg.opType, msg.element);
+});
+```
+
+# Sync with the cloud
+
+Once you're happy with your application, you can start syncing with the cloud by provisioning a backend from your project. This can be done using the `amplify-app` npx script or with the Amplify CLI. Provisioning will also create a project in the [AWS Amplify Console](https://aws.amazon.com/amplify/console/) to store metadata (such as the GraphQL schema) which you can pull down to generate models on other platforms.
+
+DataStore can connect to an existing AppSync backend that has been deployed from another JavaScript project or even it was originally deployed by iOS or Android. In these workflows it is best to work with the CLI directly by running an `amplify pull` command from your terminal and then generating models, either using `npm run amplify-modelgen` from the NPX script or with `amplify codegen models` using the Amplify CLI.
+
+For more information on this workflow please see the [Multiple Frontends documentation](https://aws-amplify.github.io/docs/cli-toolchain/quickstart#multiple-frontends){:target="_blank"}.
+
+## Use NPX
+```
+npm run amplify-push
+```
+
+If you do not already have a local AWS profile with credentials (automatically setup with the Amplify CLI) you will be prompted to do this on the first push. 
+{: .callout .callout--info}
+
+## Use Amplify CLI
+```
+amplify push
```

Comment:
We do need a small doc update here for the just adding a step on the "if you get an error with the push remove the default profile"